### ISSUE TITLE
feature request: support for AWS Batch Array Jobs

### ISSUE THREAD

**Issue by dtenenba** [broadinstitute/cromwell#4707]
If cromwell could support [Array Jobs](https://docs.aws.amazon.com/batch/latest/userguide/array_jobs.html) on AWS, that would be awesome.

It would be ideal for scatters and would hopefully resolve rate limiting issues as described in #4496.

Hoping discussion about this feature can be tracked here.

**Comment by geoffjentry** [comment]
That's actually pretty tough to do with the way Cromwell's backend interface is currently architected. We've run into this with HPC clusters as well. By the time something makes it to the backend interface it's already at the granularity of a single shard, and it's only on the **other** side of the backend interface that the code can be aware of the potential concept of an array job.

It's certainly not an insurmountable problem (it's just code) but it'd be a fairly major architectural shift, and we know that there are users who have private backend implementations and thus we try not to change the interface as much as possible

**Comment by dtenenba** [comment]
Gotcha. It shouldn't really matter except for the side effects that *not* using array jobs can cause (as referenced above).

**Comment by geoffjentry** [comment]
Yeah definitely. If we hit an impasse we'll have to figure out how to proceed. Here's hoping that we don't hit an impasse :)
